src/MongoDB.Driver.Core/ReadPreferenceHedge.cs

```diff
@@ -0,0 +1,142 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using MongoDB.Bson;
+
+namespace MongoDB.Driver
+{
+    /// <summary>
+    /// Represents the read preference hedge.
+    /// </summary>
+    public abstract class ReadPreferenceHedge : IEquatable<ReadPreferenceHedge>
```

Comment:
I considered an enum but an enum has no potential for future capabilities, so I decided to go with classes.

For example, even though `CustomReadPreferenceHedge` only takes a `bool` today I have seen discussions considering other configurable values like `delay` and `n` (n to control the degree of fanout).

Reply:
Yes, options are likely coming in future releases if the feature is deemed a success.

Reply:
But I'm not clear of the advantages of making this class abstract and adding subclasses.  Do you anticipate any additional subclasses beyond the two already defined below?

Reply:
I've removed the abstract class at your request.

If we ever have new kinds of hedges we will have a design problem.

For example, right now we have:

    var hedge = new ReadPreferenceHedge(true);

but what if in the future the server lets us specify an integer:

    var hedge = new ReadPreferenceHedge(3); 

If we add such a constructor now we have ONE class that has to represent TWO different kind of hedges.